src/LinearSolvers/preconditioners.jl

```diff
@@ -300,10 +269,35 @@ Note that this implements the same algorithm as for the GPU and becomes very slo
 
 """
 function update(p, J::SparseMatrixCSC, device)
-    kernel! = update_cpu_kernel!(device)
     p.P.nzval .= 0.0
-    ev = kernel!(J.colptr, J.rowval, J.nzval, p, p.lpartitions, ndrange=p.nblocks, dependencies=Event(device))
-    wait(ev)
+    # Fill Block Jacobi
+    # TODO: Enabling threading leads to a crash here
+    for b in 1:p.nblocks
```

Comment:
Looks good to me. I think this code can be optimized further on the CPU, but that's not the scope of this PR. Maybe should we open an issue about this?